src/containers/trending-underground/TrendingUndergroundPage.tsx

```diff
@@ -0,0 +1,137 @@
+import DesktopHeader from 'components/general/header/desktop/Header'
+import Page from 'components/general/Page'
+import Lineup from 'containers/lineup/Lineup'
+import { useDispatch } from 'react-redux'
+import { isMobile } from 'utils/clientUtil'
+import { useFlag } from 'containers/remote-config/hooks'
+import { FeatureFlags, getFeatureEnabled } from 'services/remote-config'
+import { useNavigateToPage } from 'hooks/useNavigateToPage'
+import {
+  BASE_URL,
+  NOT_FOUND_PAGE,
+  TRENDING_UNDERGROUND_PAGE
+} from 'utils/route'
+import { useMobileHeader } from 'components/general/header/mobile/hooks'
+import MobilePageContainer from 'components/general/MobilePageContainer'
+import styles from './TrendingUndergroundPage.module.css'
+import RewardsBanner from 'containers/discover-page/components/RewardsBanner'
+
+const useTrendingUndergroundLineup = (containerRef: HTMLElement) => {
+  return useLineupProps({
+    actions: trendingUndergroundLineupActions,
+    getLineupSelector: getLineup,
+    variant: LineupVariant.MAIN,
+    scrollParent: containerRef,
+    rankIconCount: 5,
+    isTrending: true,
+    isOrdered: true
+  })
+}
+
+const messages = {
+  trendingUndergroundTitle: 'Underground Trending',
+  description: "Listen to what's trending on the Audius platform"
+}
+
+type TrendingUndergroundPageProps = {
+  containerRef: HTMLElement
+}
+
+const MobileTrendingUndergroundPage = ({
+  containerRef
+}: TrendingUndergroundPageProps) => {
+  const lineupProps = useTrendingUndergroundLineup(containerRef)
+  useMobileHeader({ title: messages.trendingUndergroundTitle })
+  return (
+    <MobilePageContainer
+      title={messages.trendingUndergroundTitle}
+      description={messages.description}
+      canonicalUrl={`${BASE_URL}${TRENDING_UNDERGROUND_PAGE}`}
+      hasDefaultHeader
+    >
+      <div className={styles.mobileLineupContainer}>
+        <div className={styles.mobileBannerContainer}>
+          <RewardsBanner bannerType='underground' />
+        </div>
+        <Lineup {...lineupProps} />
+      </div>
+    </MobilePageContainer>
+  )
+}
+
+const DesktopTrendingUndergroundPage = ({
+  containerRef
+}: TrendingUndergroundPageProps) => {
+  const lineupProps = useTrendingUndergroundLineup(containerRef)
+
+  const header = (
+    <DesktopHeader primary={messages.trendingUndergroundTitle} variant='main' />
+  )
+
+  return (
+    <Page
+      title={messages.trendingUndergroundTitle}
+      description={messages.description}
+      size='large'
+      header={header}
+    >
+      <div className={styles.bannerContainer}>
+        <RewardsBanner bannerType='underground' />
+      </div>
+      <Lineup {...lineupProps} />
+    </Page>
+  )
+}
+
+const useLineupReset = () => {
+  const dispatch = useDispatch()
+  useEffect(() => {
+    return () => {
+      dispatch(trendingUndergroundLineupActions.reset())
+    }
+  }, [dispatch])
+}
+
+export const useIsTrendingUndergroundEnabled = () => {
```

Comment:
I wonder if we should define these hooks nearer to useFlag. might be easier to find all the uses of a particular flag when going through remove later on

Reply:
Hmm interesting - my gameplan for removing this was to just delete the hook and then see all the places where it fails to compile. I don't super love the idea of the remote config folder growing as we add more features, beyond the need to have defaults etc. Think I'll stick with colocating for now

Reply:
colocating is totally fine w/ me esp if you're the one to remove it. i think that if someone else becomes the one to gut the flags later, having them colocated is easier to trace